core/arch/arm/kernel/generic_boot.c

```diff
@@ -775,6 +775,12 @@ static int add_res_mem_dt_node(struct dt_descriptor *dt, const char *name,
 		if (len_size < 0)
 			return -1;
 	} else {
+		/* if there is no /reserved-memory node, retrieve address-cells
+		 * and size-cells from the root node
+		 */
+		addr_size = fdt_size_cells(dt->blob, 0);
+		len_size = fdt_address_cells(dt->blob, 0);
```

Comment:
- You got these swapped
- What if `addr_size < 0` or `len_size < 0`?

I think the calls had better be moved outside the conditional like so:
```C
        int offs = 0;
        int ret = 0;
        bool found = true;
        int addr_size = -1;
        int len_size = -1;
        char subnode_name[80];

        offs = fdt_path_offset(dt->blob, "/reserved-memory");
        if (offs < 0) {
                found = false;
                offs = 0;
        }

        addr_size = fdt_address_cells(dt->blob, offs);
        if (addr_size < 0)
                return -1;
        len_size = fdt_size_cells(dt->blob, offs);
        if (len_size < 0)
                        return -1;
        if (!found) {
                offs = add_dt_path_subnode(dt, "/", "reserved-memory");
                if (offs < 0)
                        return -1;
                ret = fdt_setprop_cell(dt->blob, offs, "#address-cells",
                                       addr_size);
                if (ret < 0)
                        return -1;
                ret = fdt_setprop_cell(dt->blob, offs, "#size-cells", len_size);
                if (ret < 0)
                        return -1;
                ret = fdt_setprop(dt->blob, offs, "ranges", NULL, 0);
                if (ret < 0)
                        return -1;
        }

        snprintf(...
```

Reply:
I'll unswap and apply your suggestion.